src/main/java/com/epam/spring/library/dto/mapper/impl/LanguageMapperImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.epam.spring.library.dto.mapper.impl;
+
+import com.epam.spring.library.dto.mapper.LanguageMapper;
+import com.epam.spring.library.model.Language;
+import com.epam.spring.library.repository.LanguageRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class LanguageMapperImpl implements LanguageMapper {
+    private final LanguageRepository repository;
+
+    @Override
+    public Language toLanguage(String code) {
+        if (code == null) {
+            return null;
+        }
+
+        return repository.getLanguage(code);
+    }
+
+    @Override
+    public String toDTO(Language language) {
+        if (Objects.isNull(language)) {
+            return null;
+        }
+
+        return language.getCode();
+    }
+
+    @Override
+    public List<String> toDTO(List<Language> languages) {
+        if (Objects.isNull(languages)) {
+            return new ArrayList<>();
+        }
+
+        return languages.stream()
+                        .map(Language::getCode)
+                        .collect(Collectors.toList());
+    }
+
+    @Override
+    public Map<String, String> toDTO(Map<Language, String> map) {
+        if (Objects.isNull(map)) {
+            return new HashMap<>();
```

Comment:
The same3 as with `new ArrayList()` -> use `Collections.emptyMap()` or `Map.of()`